### PR TITLE
feat(mobile): Create EditTransactionScreen with delete (#214)

### LINKED ISSUE #214: mobile: Create EditTransactionScreen with delete
## Problem
No way to edit or delete transactions from mobile.

### Screen: EditTransactionScreen
Navigate from TransactionListItem tap

### Form Fields (pre-populated)
- Type toggle: Income / Expense
- Amount input
- Category picker
- Date picker
- Description

### Actions
- Save: calls transactionsStore.updateTransaction
- Delete: confirmation dialog, then transactionsStore.deleteTransaction
- Cancel: dismisses without saving

### Navigation
- From TransactionsScreen item tap
- From DashboardScreen recent transaction tap

## Depends on
- transactionsStore (exists)
- #212 AddTransactionScreen (share form components)

## Acceptance Criteria
- [ ] Pre-populates with existing data
- [ ] Updates successfully
- [ ] Delete with confirmation
- [ ] Loading states
- [ ] Error handling

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

Implements EditTransactionScreen for editing and deleting transactions from the mobile app.

**Key Features:**
- Pre-populated form with existing transaction data (type, amount, category, date, description)
- Update transaction functionality with validation
- Delete transaction with confirmation dialog
- Navigation from TransactionsScreen and DashboardScreen
- Loading states and error handling
- Comprehensive test suite (50+ test cases)

**Changes:**
- Created `EditTransactionScreen.tsx` - Main edit screen component
- Created comprehensive test file with 50+ test cases
- Registered screen in navigation stack (modal presentation)
- Wired navigation from TransactionsScreen (tap on transaction)
- Wired navigation from DashboardScreen (tap on recent transaction)
- Updated mobile/README.md and CHANGELOG.md

## Test Plan

**Automated:**
- ✓ 50+ test cases covering rendering, pre-population, update, delete, validation
- ✓ All edge cases tested (confirmation cancel, error states, validation)
- ✓ Type checking passes
- ✓ Code review approved (0 critical/high issues)

**Manual Testing:**
1. Navigate to TransactionsScreen, tap a transaction
2. Verify EditTransactionScreen opens with correct pre-populated data
3. Edit amount/description/category, tap "Update Transaction"
4. Verify update persists in TransactionsScreen
5. Open edit again, tap "Delete Transaction"
6. Verify confirmation dialog appears with clear warning
7. Confirm delete, verify transaction removed from list
8. Test from DashboardScreen recent transactions

## Related Issues

Closes #214

### CODE REVIEW

File: mobile/src/screens/main/EditTransactionScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,1052 @@
...
+
+    try {
+      await updateTransaction({
+        id: transactionId,
+        categoryId: categoryId!,
+        type,
+        amount: parseFloat(amount),
+        currency,
+        date: formatDateToLocalISO(date),
+        description: description.trim() || undefined,
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply mobile/src/screens/main/EditTransactionScreen.tsx]: ✓ Fixed in d493976. Changed to description.trim() to allow users to clear the description field by sending an empty string.

File: mobile/README.md [hunk 2]
```diff
@@ -470,8 +471,56 @@ Modal screen for creating new transactions with comprehensive form:
...
+**User Experience:**
+- Confirmation dialog before deletion
+- Clear visual feedback for destructive actions
+- Categories automatically filtered when type changes
+- Character counter for description field
+
 ### Validation Utilities
 
+
+
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply mobile/README.md]: ✓ Fixed in d493976. Removed the extra blank lines in the Validation Utilities section.
